src/com/gfredericks/like_format_but_with_named_args.clj

```diff
@@ -46,11 +53,17 @@
                    ;; "%%" -> "%" conversion.
                    (format clojure-format-string)
                    `(let [arg-map# ~arg-map]
-                      (apply format ~clojure-format-string (map #(get arg-map# %) ~names)))))
+                      (try (apply format ~clojure-format-string (map #(get arg-map# %) ~names))
+                           (catch java.util.IllegalFormatConversionException e#
+                             (throw-on-bad-format ~clojure-format-string arg-map# e#))))))
```

Comment:
unquoting `clojure-format-string` a second time here has the slight drawback that the string literal will be compiled into the code twice; this can be avoided by unquoting it once in the `let` form on line 48/55: `clojure-format-string# ~clojure-format-string`, the same way as `arg-map`.